### PR TITLE
fix: Public Schedule Mobile View: change in page menu

### LINKED ISSUE #5899: Public Schedule Mobile View: change in page menu
The in-page menu has a burger menu that is the same as the top burger menu. It should be different.

Please implement a kind of drop-down button with an icon for each menu item and a down arrow. Take inspiration from GitHub as in below screenshot, e.g. see the button "development". In contrast to Github make the menu into a dropdown menu instead of a pop-up menu.

Screenshot_20201204_081234_com android chrome
Screenshot_20201204_081437_com android chrome

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes #5899

#### Short description of what this resolves:
Implemented a kind of drop-down button with an icon for each menu item and a down arrow.
The dropdown gets expanded by the icon on the right hand side.
So, if the user is on the page "info", menu shows info. If the user wants to go to ticket, open menu, scroll to Tickets and then collapse the menu.

SCREENSHOTS -

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by MeghalBisht** [comment]
One fix left, once we click on any menu item. it should hide from the menu and display on top and subsequently any previous hidden item should be displayed back. Pls check the rest @mariobehling @iamareebjamal  so I can proceed with it. Thanks

**Review by daretobedifferent18 — COMMENTED** [review]
Steps | Error
---------- | -----------------
Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280/speakers (Not showing Page Name correctly) | Screenshot from 2020-12-11 19-59-51
Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280. Click on Speakers and then on Sponsors (Not scrolling to specific section)  | Screenshot from 2020-12-11 20-03-13

**Comment by MeghalBisht** [comment]
> Steps	Error
> Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280/speakers (Not showing Page Name correctly)	Screenshot from 2020-12-11 19-59-51
> Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280. Click on Speakers and then on Sponsors (Not scrolling to specific section)	Screenshot from 2020-12-11 20-03-13

2 is fixed now, working on 1

File: app/components/public/side-menu.js [hunk 1]
```diff
@@ -60,24 +62,39 @@ export default class SideMenu extends Component {
...
+    this.set('activeMenuItem', menuOption);
     document.querySelector(`#${section}`).scrollIntoView({
       behavior: 'smooth'
-    });
+    }, this.toggleProperty('isMenuOpen'));
     this.set('activeSection', null);
     document.querySelectorAll('.scroll').forEach(node => {
       node.classList.remove('active');
     });
-    document.querySelector(`[href='#${section}']`).classList.add('active');
```
Comment (daretobedifferent18) [inline app/components/public/side-menu.js:76]: why removed this?

**Review by daretobedifferent18 — COMMENTED** [review]
At present this tab is only clickable when we click on the arrow down icon.
Screenshot from 2020-12-12 01-13-18

Make this possible wherever we click on tab

**Comment by MeghalBisht** [comment]
Thanks @daretobedifferent18, a lot of changes are required, I will update it

**Comment by MeghalBisht** [comment]
> Clicking on any tab is giving doing scrolling ok but the scrolling stops at the wrong position.
> Click on foreg. sponsors and see the error

This is happening because the menu is collapsed after the scrolling is done and hence overall height is changed.
And menu cannot be collapsed before scrolling, so should the menu be taken out from the document flow?

**Comment by MeghalBisht** [comment]
> Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280/speakers (Not showing Page Name correctly)!

Not fixed yet.

**Review by daretobedifferent18 — COMMENTED** [review]
> Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280/speakers (Not showing Page Name correctly)

Not fixed

**Comment by MeghalBisht** [comment]
> > Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280/speakers (Not showing Page Name correctly)
> 
> Not fixed

@daretobedifferent18 I was not able to do this, can you please suggest ways to approach this.

**Comment by daretobedifferent18** [comment]
> > > Goto https://open-event-frontend-ok4aqi1dy.vercel.app/e/ae73f280/speakers (Not showing Page Name correctly)
> > 
> > 
> > Not fixed
> 
> @daretobedifferent18 I was not able to do this, can you please suggest ways to approach this.

check for current route and add active class to menu option accordingly

**Comment by MeghalBisht** [comment]
And is everything else working fine? Pls check once, thanks

**Comment by iamareebjamal** [comment]
These are way too many changes. There must be a better way

**Comment by MeghalBisht** [comment]
@iamareebjamal Mobile view needs a different menu so a dropdown menu is created, the code is repeating for side-menu (in my PR), I can reduce that, will that work?

**Comment by iamareebjamal** [comment]
Yes, please

**Comment by MeghalBisht** [comment]
@iamareebjamal Please check now

**Review by daretobedifferent18 — COMMENTED** [review]
Actual | Expected
-------- | --------
Screenshot from 2020-12-15 23-35-49 | Screenshot from 2020-12-15 23-35-59


Going to specific page should show page name along with icon

**Comment by daretobedifferent18** [comment]
@MeghalBisht Still there is a lot of code that is required to solve this issue.
I think there could be a simpler and optimal solution to it.

**Comment by MeghalBisht** [comment]
> Going to specific page should show page name along with icon

That menu item is preselected and is not actually chosen by the user so to show the difference it is muted.

**Comment by MeghalBisht** [comment]
> @MeghalBisht Still there is a lot of code that is required to solve this issue.
> I think there could be a simpler and optimal solution to it.

About 80 insertions are done to add the dropdown menu, rest is refactoring. Please point out the extra code, thankyou

**Comment by MeghalBisht** [comment]
@mariobehling @iamareebjamal  Please review.

**Comment by iamareebjamal** [comment]
But why is there a need for such insertions?

**Comment by iamareebjamal** [comment]
I'll be having a quick technical meeting tomorrow hopefully, please join it to discuss better solutions and the need for adding extra code

**Comment by MeghalBisht** [comment]
@iamareebjamal A collapsible dropdown menu with icons was requested in the issue. If I modify the current menu, then more insertions would be required because styles, dropdown icon, hiding/unhiding menu items would be changed conditionally for mobile view which will make it even more complex.

Also, the new menu behaves better than the previous one.
1. If I switch pages in the old menu I cannot sense the switch unless I scroll down.
2.  Every time I need to collapse the menu manually
3. If I skip the burger icon, I have no way to switch the pages.

**Comment by MeghalBisht** [comment]
@iamareebjamal  Please check now, I have refactored most of the code

**Comment by iamareebjamal** [comment]
Code looks good to me now. @maze-runnar @sachinchauhan2889 @daretobedifferent18 Please review now

**Review by daretobedifferent18 — COMMENTED** [review]
clicking on foreg tickets and then clicking on info gives an error. Please fix that

**Comment by maze-runnar** [comment]
it's working fine as far I checked. Only thing that when you navigate away from page and then go back, then it shows select selection. if it's ok then evrything else looks good to me.
open-event-frontend-git-fork-meghalbisht-schedulemb-5899 eventyay now sh_e_a315169f(Moto G4)

**Comment by daretobedifferent18** [comment]
> it's working fine as far I checked. Only thing that when you navigate away from page and then go back, then it shows select selection. if it's ok then evrything else looks good to me.
> open-event-frontend-git-fork-meghalbisht-schedulemb-5899 eventyay now sh_e_a315169f(Moto G4)

Yes working fine for me as well

**Comment by MeghalBisht** [comment]
> clicking on foreg tickets and then clicking on info gives an error. Please fix that

Please tell the exact steps or add a screenshot of the error as I cannot reproduce it

**Comment by daretobedifferent18** [comment]
> > clicking on foreg tickets and then clicking on info gives an error. Please fix that
> 
> Please tell the exact steps or add a screenshot of the error as I cannot reproduce it

I have rechecked it. Its working fine for me now

**Comment by MeghalBisht** [comment]
The only problem left is -
If I go to one link lets say /speakers and then go to another /calendar.
Now, if I press back then 
Expected : I should see speakers but instead I see calendar! 
I am not sure how to fix this, any inputs will be helpful

**Comment by iamareebjamal** [comment]
Was it there previously?

**Comment by MeghalBisht** [comment]
No, only in the UiDropdown I added.

**Comment by daretobedifferent18** [comment]
> The only problem left is -
> If I go to one link lets say /speakers and then go to another /calendar.
> Now, if I press back then
> Expected : I should see speakers but instead I see calendar!
> I am not sure how to fix this, any inputs will be helpful

I am not getting this error. I think the error I talked about going back to info was also there but is not showing now.
@iamareebjamal i think there might be some other error related problem where error sometimes comes and sometimes not.

**Comment by MeghalBisht** [comment]
@daretobedifferent18  Can you describe what was the error about, as I still cannot reproduce it

**Comment by iamareebjamal** [comment]
Merging for now

**Comment by MeghalBisht** [comment]
@iamareebjamal Please confirm are you able to reproduce this https://github.com/fossasia/open-event-frontend/pull/5998#issuecomment-749547605 ?

**Comment by mariobehling** [comment]
Ok, so the change here also shows the icons on the sidebar of the desktop view. There should be no icon in the desktop.

**Comment by mariobehling** [comment]
I opened an issue here https://github.com/fossasia/open-event-frontend/issues/6118

**Comment by MeghalBisht** [comment]
Okay, I thought they were needed and hence added on purpose, I will follow up!

**Comment by iamareebjamal** [comment]
> @iamareebjamal Please confirm are you able to reproduce this [#5998 (comment)](https://github.com/fossasia/open-event-frontend/pull/5998#issuecomment-749547605) ?

No

**Pull request merged**